voice_assistant_pkg/voice_assistant_pkg/speech_recognizer.py

```diff
@@ -0,0 +1,72 @@
+import os
+import openai
+import speech_recognition as sr
+
+class SpeechRecognizer:
+    def __init__(self, mic_index, energy_threshold, timeout, phrase_time_limit, use_whisper=True, dynamic_energy_threshold=False):
+        self.speech_recognizer = sr.Recognizer()
+        self.mic = sr.Microphone(device_index=mic_index) # 11 or 12: self microphone, 13: snowball
+        self.speech_recognizer.dynamic_energy_threshold=dynamic_energy_threshold
+        self.speech_recognizer.energy_threshold =energy_threshold
+        self.use_whisper=use_whisper
+        self.timeout=timeout
+        self.phrase_time_limit=phrase_time_limit
+
+        openai.api_key = os.environ.get('OPENAI_API_KEY')
+
+    def whisper(self, audio):
+        """Transcribe audio using openAI's whisper model 
+
+        Args:
+            audio : Speech recognizer object that contains user's input 
+
+        Returns:
+            user_input (str): transcribed voice command
+        """
+        with open('speech.wav','wb') as f:
+            f.write(audio.get_wav_data())
+        speech = open('speech.wav', 'rb')
```

Comment:
Will this be saved inside the workspace?

Reply:
yes, where else should we save it? Should I create a user-defined path parameter to save it?

Reply:
This is fine for now, but probably better to save it to a temp directory unless we want to keep it around long-term